src/platforms/windows/daemon/windowssplittunnel.cpp

```diff
@@ -0,0 +1,535 @@
+        }
+    }
+
+
+}
+
+
+std::vector<uint8_t> WindowsSplitTunnel::generateProcessBlob(){
+
+    // Get a Snapshot of all processes that are running:
+    HANDLE snapshot_handle = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
+    if (snapshot_handle == INVALID_HANDLE_VALUE){
+        WindowsCommons::windowsLog("Creating Process snapshot failed");
+        return std::vector<uint8_t>(0);
+    }
+    // Load the First Entry, later iterate over all
+    PROCESSENTRY32W currentProcess;
+    currentProcess.dwSize = sizeof(PROCESSENTRY32W);
+
+    if (FALSE ==(Process32First(snapshot_handle, &currentProcess)))
+    {
+        WindowsCommons::windowsLog("Cant read first entry");
+    }
+
+    QMap<DWORD, ProcessInfo> processes;
+
+    do{
+        auto process_handle = OpenProcess(
+                    PROCESS_QUERY_LIMITED_INFORMATION, FALSE, currentProcess.th32ProcessID);
+
+        if(process_handle == INVALID_HANDLE_VALUE){
+            continue;
+        }
+        ProcessInfo info = getProcessInfo(process_handle,currentProcess);
+        processes.insert(info.ProcessId,info);
+
+    }while (FALSE != (Process32NextW(snapshot_handle, &currentProcess)));
+
+
+    auto process_list = processes.values();
+    if( process_list.isEmpty()){
+        logger.log() << "Process Snapshot list was empty";
+        return std::vector<uint8_t>(0);
+    }
+
+    logger.log() << "Reading Processes NUM: " << process_list.size();
+    //Determine the Size of the outBuffer:
+    size_t totalStringSize = 0;
+
+    for (const auto &process : process_list)
+    {
+        totalStringSize += (process.DevicePath.size() * sizeof(wchar_t));
+    }
+    auto bufferSize = sizeof(PROCESS_DISCOVERY_HEADER)
+        + (sizeof(PROCESS_DISCOVERY_ENTRY) * processes.size())
+        + totalStringSize;
+
+    std::vector<uint8_t> out(bufferSize);
+
+    auto header = reinterpret_cast<PROCESS_DISCOVERY_HEADER*>(&out[0]);
+    auto entry = reinterpret_cast<PROCESS_DISCOVERY_ENTRY*>(header + 1);
+    auto stringBuffer = reinterpret_cast<uint8_t *>(entry + processes.size());
+
+    SIZE_T currentStringOffset = 0;
+
+    for (const auto &process : process_list)
+    {
+        // Wierd DWORD -> Handle Pointer magic.
+        entry->ProcessId = (HANDLE)((size_t) process.ProcessId);
+        entry->ParentProcessId =(HANDLE)((size_t) process.ParentProcessId);
+
+        if (process.DevicePath.empty())
+        {
+            entry->ImageNameOffset = 0;
+            entry->ImageNameLength = 0;
+        }else{
+            const auto imageNameLength = process.DevicePath.size() * sizeof(wchar_t);
+
+            entry->ImageNameOffset = currentStringOffset;
+            entry->ImageNameLength = static_cast<USHORT>(imageNameLength);
+
+            RtlCopyMemory(stringBuffer + currentStringOffset, &process.DevicePath[0], imageNameLength);
+
+            currentStringOffset += imageNameLength;
+        }
+        ++entry;
+    }
+
+    header->NumEntries = processes.size();
```

Comment:
It seems that you use `processes` only here and it should have the same size of `process_list`. Wonder if you can use a smile QList<> instead of a QMap<>.